django/db/models/sql/compiler.py

```diff
@@ -1836,7 +1836,23 @@ def pre_sql_setup(self):
         query.clear_ordering(force=True)
         query.extra = {}
         query.select = []
-        query.add_fields([query.get_meta().pk.name])
+        meta = query.get_meta()
+        fields = [meta.pk.name]
+        related_ids_index = []
+        for related in self.query.related_updates:
+            if all(
+                path.join_field.primary_key for path in meta.get_path_to_parent(related)
+            ):
+                # If a primary key chain exists to the targeted related
+                # update the `meta.pk` value can be used for it.
+                related_ids_index.append((related, 0))
+            else:
+                # This branch will only be reached when updating a field
+                # of an ancestor not that is not part of the primary key
+                # chain of a MTI tree.
+                related_ids_index.append((related, len(fields)))
+                fields.append(related._meta.pk.name)
```

Comment:
When MTI is used and an attempt is made at updating a field accessible through a parent link that isn't in the primary key chain (since Django doesn't support composite primary keys the primary of a child with multiple parents is a one-to-one to the first parent) then the value is selected from the first `SELECT` by relying on joins.